internal/pkg/agent/application/managed_mode.go

```diff
@@ -182,32 +183,26 @@ func (m *managedConfigManager) Run(ctx context.Context) error {
 	}
 
 	// Proxy errors from the gateway to our own channel.
-	go func() {
+	gatewayErrorsRunner := runner.Start(context.Background(), func(ctx context.Context) error {
 		for {
 			select {
 			case <-ctx.Done():
-				return
+				return nil
 			case err := <-gateway.Errors():
 				m.errCh <- err
 			}
 		}
-	}()
+	})
+	defer gatewayErrorsRunner.Stop()
 
 	// Run the gateway.
-	gatewayRun := make(chan bool)
-	gatewayErrCh := make(chan error)
-	defer func() {
-		gatewayCancel()
-		<-gatewayRun
-	}()
-	go func() {
-		err := gateway.Run(gatewayCtx)
-		close(gatewayRun)
-		gatewayErrCh <- err
-	}()
+	gatewayRunner := runner.Start(gatewayCtx, func(ctx context.Context) error {
+		defer gatewayErrorsRunner.Stop()
```

Comment:
should this be `gatewayRunner.Stop()` ?

Reply:
nope, the errors runner should be stopped only after the gateway run stopped

Reply:
we could and yeah, it's safe as calling context cancel multiple times